### PR TITLE
Add timeout on wait_for_hosts_stage

### PR DESCRIPTION
**Pull request by iovadia**
Adding timeout variable for waiting to all host to reach progress stage

Before this PR we were limited by the default  CLUSTER_INSTALLATION_TIMEOUT / 2

### CODE REVIEW

**Comment by iovadia** [comment]
cc @eliorerz

File: src/assisted_test_infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -588,13 +588,21 @@ def wait_for_non_bootstrap_masters_to_reach_joined_state_during_install(self, nu
             nodes_count=num_masters - 1,
         )
 
-    def wait_for_hosts_stage(self, stage: str, inclusive: bool = True):
+    def wait_for_hosts_stage(
+        self,
+        stage: str,
+        inclusive: bool = True,
+        timeout=consts.CLUSTER_INSTALLATION_TIMEOUT / 2,
+        interval=consts.DEFAULT_CHECK_STATUSES_INTERVAL,
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/helper_classes/cluster.py:596]:
```suggestion
        timeout: int = consts.CLUSTER_INSTALLATION_TIMEOUT // 2,
        interval: int = consts.DEFAULT_CHECK_STATUSES_INTERVAL,
```

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/test_infra/helper_classes/cluster.py (hunk 1 above)
Comment (iovadia) [reply src/assisted_test_infra/test_infra/helper_classes/cluster.py]: done, thanks

File: src/assisted_test_infra/test_infra/helper_classes/cluster.py [hunk 2]
```diff
@@ -588,13 +588,21 @@ def wait_for_non_bootstrap_masters_to_reach_joined_state_during_install(self, nu
...
+        interval: int = consts.DEFAULT_CHECK_STATUSES_INTERVAL,
+    ):
         index = consts.all_host_stages.index(stage)
         utils.waiting.wait_till_at_least_one_host_is_in_stage(
             client=self.api_client,
             cluster_id=self.id,
             stages=consts.all_host_stages[index:] if inclusive else consts.all_host_stages[index + 1 :],
             nodes_count=self.nodes.nodes_count,
+            timeout=timeout,
+            interval=interval,
```
[the remark this answers is not in the document]
Comment (iovadia) [reply src/assisted_test_infra/test_infra/helper_classes/cluster.py]: Done

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**